### DISCUSSION TITLE
Migrating from 7.9.2 to 8.x.x

### DISCUSSION THREAD

**Discussion by ChristosLabrou** [gotenberg/gotenberg#837] [Q&A] [answered]
I am trying to replicate setting the `uno-converter-restart-threshold=0` flag from 7.9.2 to 8.x.x. 

I'm having performance issues so I'm asking here to determine if they're due to incorrect settings or due to some of the issues already posted. Based on documentation the equivalent setting should be `libreoffice-restart-after=1` correct?

**Comment by gulien** [comment, marked as answer]
Almost 😄 `libreoffice-restart-after=0`

**Reply by gulien** [reply]
With 0, LibreOffice will never restart tough. According to your requirements, you may consider restarting every 10 to more conversions to prevent any memory leaks.

With your current configuration, you’re LibreOffice instance is restarted every conversion, which explains the performance issues.

**Reply by manoskouvarakis** [reply]
We are experiencing the same issue and we need your help @gulien. 
We had `uno-converter-restart-threshold=0` that from documentation means "Stateless a dedicated LibreOffice instance per conversion".
Moving to v8.14 we believed that in order to have a stateless libreoffice we need to set `libreoffice-restart-after=1` because if we set it to 0, it will never restart. 
Is `1` the equivalent to what we already had (the stateless option)?

**Reply by gulien** [reply]
@manoskouvarakis 1 means it will restart every conversion, but there is only one instance instead of many (there is a queue mechanism). If you want // conversions, you have to spin more Gotenberg instances.

**Reply by manoskouvarakis** [reply]
Is there any way we can have many libreoffice processes in one gotenberg instance, that each of them restarts after every conversion? (from what I understand this was the case with v7.9.x).

Also is there any difference on option `--libreoffice-max-queue-size=0` vs `--libreoffice-max-queue-size=1`?

**Reply by gulien** [reply]
1. Nope, it's not the role of your container orchestrator.
2. `--libreoffice-max-queue-size=1` = only one request processed by LibreOffice, and in the meantime all requests using LibreOffice are cancelled early.
